test.py

```diff
@@ -563,16 +563,65 @@ def run_test(framework, cluster, repo_root):
         logger.info(msg, runtests_script, framework.name)
         raise CommandFailure(cmd_args)
 
+def setup_junit_xml():
+    """Try to download and import junit_xml"""
+    workdir = get_work_dir()
+    lib_dir = os.path.join(workdir, 'lib', 'python')
+    pip_cmd = ['pip', 'install', 'junit-xml', '-t', lib_dir]
```

Comment:
> Are you objecting to the pattern of a process downloading and loading a module itself, or just looking for different module organization?

Kind of, yeah. I meant this in the most simple sense possible: setting up the environment (here, installing Python modules) is an orthogonal concern to something like emitting junit.xml output. Keeping these two concerns grouped together in different files/modules makes them easier to find and modify.